ts/components/messages/hooks/useMessagesFetcher.tsx

```diff
@@ -0,0 +1,116 @@
+import * as pot from "@pagopa/ts-commons/lib/pot";
+import { useEffect } from "react";
+import I18n from "../../../i18n";
+import { maximumItemsFromAPI, pageSize } from "../../../config";
+import {
+  loadNextPageMessages,
+  loadPreviousPageMessages,
+  reloadAllMessages
+} from "../../../store/actions/messages";
+import { useIODispatch, useIOSelector } from "../../../store/hooks";
+import { Collection } from "../../../store/reducers/entities/messages/allPaginated";
+import { GlobalState } from "../../../store/reducers/types";
+import { showToast } from "../../../utils/showToast";
+import { useOnFirstRender } from "../../../utils/hooks/useOnFirstRender";
+
+export const useMessagesFetcher = (
+  archived: boolean,
+  messagesSelector: (state: GlobalState) => Collection["data"],
+  isLoadingNextPageSelector: (state: GlobalState) => boolean,
+  isLoadingPreviousPageSelector: (state: GlobalState) => boolean,
+  isReloadingAllSelector: (state: GlobalState) => boolean
+) => {
+  const dispatch = useIODispatch();
+  const messagesPot = useIOSelector(messagesSelector);
+  const isError = pot.isError(messagesPot);
+  const isSome = pot.isSome(messagesPot);
+
+  const { messages, nextCursor, previousCursor } = pot.getOrElse(
+    pot.map(messagesPot, mp => ({
```

Comment:
I'm biased towards explicit variables names (even if they are very long). Should be rename "mp" to something more meaningful here?